src/Sylius/Bundle/LocaleBundle/Form/DataTransformer/ArrayGregorianToCalendarSystemTransformer.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\LocaleBundle\Form\DataTransformer;
+
+use Sylius\Bundle\LocaleBundle\Templating\Helper\LocaleHelper;
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+/**
+ * @author Aram Alipoor <aram.alipoor@gmail.com>
+ */
+class ArrayGregorianToCalendarSystemTransformer implements DataTransformerInterface {
+
+    /**
+     * @var LocaleHelper
+     */
+    private $localeHelper;
+
+    /**
+     * Constructor.
+     *
+     * @param $localeHelper
+     */
+    public function __construct(LocaleHelper $localeHelper)
+    {
+        $this->localeHelper = $localeHelper;
+    }
+
+    /**
+     * Transforms a date array into a calendar-aware localized date array.
```

Comment:
I was thinking about comment for this and below method... Isn't that obvious what are this functions for, if you look at their names and class name? But maybe just for me, so I don't insist ;)